pyqtorch/parametric.py

```diff
@@ -395,6 +475,23 @@ def __init__(
         """
         super().__init__("X", control, target, param_name)
 
+    @cached_property
+    def eigenvals_generator(self) -> Tensor:
+        """Get eigenvalues of the underlying generator.
+
+        Arguments:
+            values: Parameter values.
+
+        Returns:
+            Eigenvalues of the generator operator.
+        """
+        return torch.cat(
+            (
+                torch.zeros(2 ** len(self.qubit_support) - 2),
```

Comment:
be very careful when instantiating torch tensors. always use the device and dtype of the gate. tagging @jpmoutinho just in case this gets lost for me

Reply:
Gotcha @dominikandreasseitz, they are now instantiated with `self.device` and `self.dtype`